synapse/cores/common.py

```diff
@@ -2497,23 +2432,82 @@ def formTufoByProp(self, prop, valu, **props):
 
             tufo = (iden, fulls)
 
+            # Cache the tufo data now so we can avoid having a .new ephemeral
+            # property in the cache
             if self.caching:
-                # avoid .new in cache
                 cachefo = (iden, dict(fulls))
                 for p, v in fulls.items():
                     self._bumpTufoCache(cachefo, p, None, v)
 
-            # fire notification events
+            # Run any autoadd nodes we may have. In the event of autoadds being
+            # present, our subsequent node:add events are fired depth-first
+            if self.autoadd and toadds is not None:
+                self._runToAdd(toadds)
+
+            # fire the node:add events from the xact
             xact.fire('node:add', form=prop, valu=valu, node=tufo)
             xact.spliced('node:add', form=prop, valu=valu, props=props)
             xact.trigger(tufo, 'node:add', form=prop)
 
-            if self.autoadd:
-                self._runToAdd(fulls)
-
         tufo[1]['.new'] = True
         return tufo
 
+    def _pruneFulls(self, form, fulls, props, isadd=False):
+        '''
+        Modify fulls and props dicts in place to remove non-model props.
+
+        Args:
+            form (str): Form of the dictionary being operated on.
+            fulls (dict): Dictionary of full property name & valu pairs.
+            props (dict): Dictionary of property name & value pairs.
+            isadd (bool): Bool indicating if the data is newly being added or not.
+
+        Returns:
+            None
+        '''
+        splitp = form + ':'
+        for name in list(fulls.keys()):
+            if name in ('tufo:form', 'tufo:formed'):
```

Comment:
node:created ( and we may eventually need this to be a bit more dynamic... planning for node:loc :D )